src/cephadm/cephadm

```diff
@@ -2810,6 +2810,14 @@ def get_container(ctx: CephadmContext,
         envs.extend(cc.get_container_envs())
         container_args.extend(cc.get_container_args())
 
+    if daemon_type in ['rgw', CephIscsi.daemon_type]:
+        # set containers limits to unlimited for iscsi and rgw daemons as defaults
+        # (Docker 4096 / Podman 2048) prevents some app customizations from running
+        if isinstance(ctx.container_engine, Podman):
+            container_args.append('--pids-limit=-1')
+        else:
+            container_args.append('--pids-limit=0')
```

Comment:
I would introduce `set_pids_limit_unlimited` method and call it from existing rgw and iscsi branches here, as well as from `get_tcmu_runner_container` to avoid duplicating the podman vs docker logic in two places.  This way you could also supply meaningful comments (i.e. mention the number of luns for iscsi and `rgw_thread_pool_size` for rgw) instead of the rather vague "app customizations" reference.

Reply:
moved to individual function. Does the comment looks alright?